internal/shells/bash.go

```diff
@@ -0,0 +1,26 @@
+package shells
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// Executes a bash command and returns the output or error.
+func ExecuteBashCommand(command string, env map[string]string, inherit_environment_variables bool) (string, error) {
+	commandToExecute := exec.Command("bash", "-c", command)
+
+	if inherit_environment_variables {
+		commandToExecute.Env = os.Environ()
+	}
+
+	for k, v := range env {
+		commandToExecute.Env = append(commandToExecute.Env, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	out, err := commandToExecute.Output()
+	if err != nil {
+		return "", fmt.Errorf("error executing bash command: %w", err)
```

Comment:
Can we pass the error to the user. Currently output looks like this:

![image](https://github.com/Azure/InnovationEngine/assets/108765066/f00476a2-11f2-4afe-81de-c31417795d9b)

Reply:
The exit code & error message from the command are now surfaced here.

Reply:
On further testing this isn't working for us:

![image](https://github.com/Azure/InnovationEngine/assets/108765066/62c567a3-9c2d-457e-a00e-a161ce17d528)

Is it that the error is going to syserr rather than sysout?

Reply:
You're right, I was only capturing the standard output. I've updated the error & tested the output to make sure that the standard error stream was captured. The error message needs proper formatting, but the error is being captured now:
![image](https://github.com/Azure/InnovationEngine/assets/6026326/c7365558-c0c4-49f9-b8af-585c79242721)

Reply:
That's good for now. Thanks you.